File: Benchmarking/src/gen_2_30.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  case class s0 (
    p0: Vector[Vector[Double]],
    p1: Vector[Vector[Double]]
  )
  case class s1 (
    p0: Vector[Vector[s0]],
    p1: s0
  )
  case class s2 (
    p0: s0,
    p1: s1
  )
  @noinline
  def f0(v0: Vector[Vector[s2]], v1: Double): Double = {
    var v6: Double = v1
    val v5: Double = v6 + v6
    val v15: Vector[Double] = Vector(v5, v6, v6, v5)
    var v9: Double = v5
    var v18: Vector[Double] = v15
    var v21: Vector[Double] = v18
    var v45: Vector[Double] = v21
    v21 = v21.updated(3, v9)
    v21 = v18
    v18 = v21
    var v69: Vector[Double] = v45
    val v78: Double = v69(2)
    v78
  }
  def benchmark(): Unit = {
    val v0: Vector[Vector[s2]] = Vector(Vector(s2(s0(Vector(Vector(0.0), Vector(1.0), Vector(2.0)), Vector(Vector(3.0), Vector(4.0))), s1(Vector(Vector(s0(Vector(Vector(5.0), Vector(6.0), Vector(7.0)), Vector(Vector(8.0), Vector(9.0))))), s0(Vector(Vector(10.0), Vector(11.0), Vector(12.0)), Vector(Vector(13.0), Vector(14.0)))))), Vector(s2(s0(Vector(Vector(15.0), Vector(16.0), Vector(17.0)), Vector(Vector(18.0), Vector(19.0))), s1(Vector(Vector(s0(Vector(Vector(20.0), Vector(21.0), Vector(22.0)), Vector(Vector(23.0), Vector(24.0))))), s0(Vector(Vector(25.0), Vector(26.0), Vector(27.0)), Vector(Vector(28.0), Vector(29.0)))))))
    val v1: Double = 30.0
    val start = nanoTime()
    var result: Double = 30.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}

File: Benchmarking/src/gen_2_30.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: [[s0]]
    var p1: s0
  }
  struct s2 {
    var p0: s0
    var p1: s1
  }
  @inline(never)
  func f0(_ v0: [[s2]], _ v1: Double) -> Double {
    var v6: Double = v1
    let v5: Double = v6 + v6
    let v15: [Double] = [v5, v6, v6, v5]
    var v9: Double = v5
    var v18: [Double] = v15
    var v21: [Double] = v18
    var v45: [Double] = v21
    v21[3] = v9
    v21 = v18
    v18 = v21
    var v69: [Double] = v45
    let v78: Double = v69[2]
    return v78
  }
  func benchmark() {
    let v0: [[s2]] = [[s2(p0: s0(p0: [[0.0], [1.0], [2.0]], p1: [[3.0], [4.0]]), p1: s1(p0: [[s0(p0: [[5.0], [6.0], [7.0]], p1: [[8.0], [9.0]])]], p1: s0(p0: [[10.0], [11.0], [12.0]], p1: [[13.0], [14.0]])))], [s2(p0: s0(p0: [[15.0], [16.0], [17.0]], p1: [[18.0], [19.0]]), p1: s1(p0: [[s0(p0: [[20.0], [21.0], [22.0]], p1: [[23.0], [24.0]])]], p1: s0(p0: [[25.0], [26.0], [27.0]], p1: [[28.0], [29.0]])))]]
    let v1: Double = 30.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 30.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
